app/controllers/buildings_controller.rb

```diff
@@ -205,8 +195,17 @@ def add_classrooms_for_building(bldrecnbr)
       end
     end
 
-    def change_rooms_archived_mode(building, archived_mode)
-      Room.where(floor_id: building.floors.ids).update_all(archived: archived_mode)
+    def change_building_archived_mode(building:, archived:)
+      ActiveRecord::Base.transaction do
+        raise ActiveRecord::Rollback unless building.update(archived: archived)
+        raise ActiveRecord::Rollback unless Room.where(floor_id: building.floors.ids).update_all(archived: archived)
+        rooms = archived ? Room.archived : Room.active
+        rooms.each do |room|
+          raise ActiveRecord::Rollback unless room.specific_attributes.update(archived: archived)
+          raise ActiveRecord::Rollback unless room.resources.update(archived: archived)
+        end
+      end
```

Comment:
won't this update ALL rooms in the db? shouldnt line 202 instead be something like this (could be written better): 
```
rooms = archived ? Room.archived.where(floor_id: building.floors.ids) : Room.active.where(floor_id: building.floors.ids)
```

or am i missing something?